components/scream/scripts/test_all_scream.py

```diff
@@ -170,29 +202,32 @@ def generate_baselines(self, test, cleanup):
             print("WARNING: Failed to create baselines:\n{}".format(err))
             return False
 
-        if cleanup:        
-            run_cmd_no_fail("ls | grep -v data | xargs rm -rf ", from_dir=test_dir)
+        run_cmd_no_fail("ls | grep -v data | xargs rm -rf ", from_dir=test_dir)
 
         return True
 
     ###############################################################################
-    def generate_all_baselines(self, git_baseline_head, git_head):
+    def generate_all_baselines(self):
     ###############################################################################
-        print("Generating baselines for ref {}".format(git_baseline_head))
+        git_head_commit     = get_current_commit()
+        git_head_ref        = get_current_head()
+        git_baseline_commit = get_current_commit(commit=self._baseline_ref)
+
+        need_co = git_baseline_commit != git_head_commit
```

Comment:
I know you have `co` as an alias of `checkout` on your machine, but perhaps we should be verbose here, so whomever sees this file next understands what is actually needed.